src/esbuild/modules.ts

```diff
@@ -21,11 +21,42 @@ export const getModulesResults = (options: LocalOptions, esbuildMeta?: Metafile)
     if (!esbuildMeta) {
         return {};
     }
+
+    // Indexing chunks so we can access them faster.
+    const outputs = esbuildMeta.outputs;
+    const chunkIndexedSets: Record<string, Set<string>> = {};
+    const chunkIndexed: Record<string, string[]> = {};
+    const parseModules = (chunkName: string, moduleName: string) => {
+        const formatedModuleName = formatModuleName(moduleName, context);
+        chunkIndexedSets[formatedModuleName] = chunkIndexedSets[formatedModuleName] || new Set();
+        const formatedChunkName = chunkName.split('/').pop()?.split('.').shift() || 'unknown';
+        chunkIndexedSets[formatedModuleName].add(formatedChunkName);
+
+        if (outputs[moduleName] && outputs[moduleName].inputs.length) {
+            for (const inputModuleName of Object.keys(outputs[moduleName].inputs)) {
+                parseModules(moduleName, inputModuleName);
+            }
+        }
+    };
+
+    for (const [chunkName, chunk] of Object.entries(outputs)) {
+        for (const moduleName of Object.keys(chunk.inputs)) {
+            parseModules(chunkName, moduleName);
+        }
+    }
+
+    for (const key of Object.keys(chunkIndexedSets)) {
+        chunkIndexed[key] = Array.from(chunkIndexedSets[key]);
+    }
+
     for (const [path, obj] of Object.entries(esbuildMeta.inputs)) {
         const moduleName = formatModuleName(path, context);
         const module: LocalModule = modulesMap[moduleName] || getDefaultLocalModule(moduleName);
 
         module.size = obj.bytes;
+        if (chunkIndexed[moduleName]) {
+            module.chunkNames = chunkIndexed[moduleName];
```

Comment:
Wouldn't the following eliminate the need for `chunIndexed` altogether?
```suggestion
            module.chunkNames = Array.from(chunkIndexed[moduleName]);
```